dts/arm/ambiq/ambiq_apollo4p.dtsi

```diff
@@ -26,6 +26,50 @@
 		reg = <0x10000000 0x60000>;
 	};
 
+	uartclk: apb-pclk {
+		compatible = "fixed-clock";
+		clock-frequency = <24000000>;
+		#clock-cells = <0>;
+	};
+
+	soc {
+		uart0: uart@4001c000 {/* UART0 */
+			compatible = "arm,pl011";
+			reg = <0x4001c000 0x1000>;
+			interrupts = <15 0>;
+			interrupt-names = "irq_15";
```

Comment:
question: is this the real IRQ name?

Reply:
Updated IRQ names to match IRQ descriptions located in the "ARM Cortex-M4 Vector Table for Apollo4 Plus SoC" table from the Apollo4 Plus Programmer’s Guide.